manim/utils/bezier.py

```diff
@@ -86,12 +88,116 @@ def bezier(
     )
 
 
-# !TODO: This function has still a weird implementation with the overlapping points
 def partial_bezier_points(points: BezierPoints, a: float, b: float) -> BezierPoints:
-    """Given an array of points which define bezier curve, and two numbers 0<=a<b<=1, return an array of the same size,
-    which describes the portion of the original bezier curve on the interval [a, b].
-
-    This algorithm is pretty nifty, and pretty dense.
+    r"""Given an array of ``points`` which define a Bézier curve, and two numbers :math:`a, b`
+    such that :math:`0 \le a < b \le 1`, return an array of the same size, which describes the
+    portion of the original Bézier curve on the interval :math:`[a, b]`.
+
+    .. seealso::
+        See :func:`split_bezier` for an explanation on how to split Bézier curves.
```

Comment:
Maybe briefly mention how it's conceptually equal to `split_bezier` twice?
That just helped me understand.

Reply:
Done!